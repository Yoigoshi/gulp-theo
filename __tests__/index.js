const gulp = require('gulp')
const through = require('through2')
const theo = require('../')

describe('gulp-theo', () => {
  it('transforms Design Tokens as Sass', (done) => {
    gulp.src('__fixtures__/a.json')
      .pipe(theo.plugin(
        {
          transform: {
            type: 'web'
          },
          format: {
            type: 'scss'
          }
        }
      ))
      .pipe(
        through.obj((file, enc, next) => {
          expect(file.contents.toString('utf8')).toMatchSnapshot()
          next()
          done()
        })
      )
  })
  it('transforms multiple files as Sass', (done) => {
    gulp.src([
      '__fixtures__/a.json',
      '__fixtures__/props.yml'
    ])
    .pipe(theo.plugin(
      {
        transform: {
          type: 'web'
        },
        format: {
          type: 'scss'
        }
      }
    ))
    .pipe(
      through.obj((file, enc, next) => {
        if (file.relative === 'a.json') {
          expect(file.contents.toString('utf8')).toMatchSnapshot()
        }
        console.log(file.relative)
        if (file.relative === 'props.scss') {
          expect(file.contents.toString('utf8')).toMatchSnapshot()
          done()
        }
        next()
      })
    )
  })
  it('throws an error if a bad configuration is passed', (done) => {
    gulp.src('__fixtures__/a.json')
    .pipe(theo.plugin(
      {
        format: {
          type: 'scss'
        }
      }
    ))
    .on('error', (e) => {
      expect(e.message).toBe('Transform "undefined" is not registered')
      done()
    })
  })

  it('transforms Design Tokens as 1337v4r5', (done) => {
    theo.registerFormat('1337v4r5', `
      ðŸ˜€{{stem meta.file}}-1337:
        {{#each props as |prop|}}
          '{{kebabCase prop.name}}': ðŸ˜‡({{prop.value}})
        {{/each}}
    `)
    gulp.src('__fixtures__/a.json')
    .pipe(theo.plugin(
      {
        transform: {
          type: 'web'
        },
        format: {
          type: '1337v4r5'
        }
      }
    ))
    .pipe(
      through.obj((file, enc, next) => {
        expect(file.contents.toString('utf8')).toMatchSnapshot()
        next()
        done()
      })
    )
  })
})
