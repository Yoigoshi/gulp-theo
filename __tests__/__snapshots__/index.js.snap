exports[`gulp-theo transforms Design Tokens as 1337v4r5 1`] = `
"
      ðŸ˜€a-1337:
          \'token-a\': ðŸ˜‡(#FF0000)
          \'token-a-a\': ðŸ˜‡(#FF0000 #FF0000)
          \'token-a-b\': ðŸ˜‡(#FFFFFF)
          \'token-same-name-as-alias\': ðŸ˜‡(foo)
          \'sizing-a\': ðŸ˜‡(20px)
          \'color-a\': ðŸ˜‡(rgb(186, 218, 85))
          \'token-b\': ðŸ˜‡(#FFFFFF)
    "
`;

exports[`gulp-theo transforms Design Tokens as Sass 1`] = `
"
  
  $token-a: #FF0000;
  
  $token-a-a: #FF0000 #FF0000;
  
  $token-a-b: #FFFFFF;
  
  $token-same-name-as-alias: foo;
  
  $sizing-a: 20px;
  
  $color-a: rgb(186, 218, 85);
  
  $token-b: #FFFFFF;
"
`;

exports[`gulp-theo transforms multiple files as Sass 1`] = `
"
  
  $foo: rgb(255, 0, 0);
  
  $baz: rgb(0, 0, 255);
"
`;
